bvm/ballerina-core/src/main/java/org/ballerinalang/util/transactions/ParticipantRegistry.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.util.transactions;
+
+import org.ballerinalang.model.values.BFunctionPointer;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+/**
+ * Registry for Participated transaction related information.
+ *
+ * @since 0.985.0
+ */
+class ParticipantRegistry {
+    Map<String, Deque<List<Participant>>> resourceParticipantReg;
+    Map<String, Map<String, Participant>> localFunctionParticipants;
```

Comment:
Shall we make them private?

Reply:
ParticipantRegistry was deprecated using failed set. This is dead code, will remove.